src/libcore/option.rs

```diff
@@ -1315,6 +1315,26 @@ impl<A, V: FromIterator<A>> FromIterator<Option<A>> for Option<V> {
     /// Since the last element is zero, it would underflow. Thus, the resulting
     /// value is `None`.
     ///
+    /// Here is a variation on the previous example, showing that no
+    /// further elements are taken from `iter` after the first `None`.
+    ///
+    /// ```
+    /// let items = vec![3_u16, 2, 1, 10];
```

Comment:
Just as a question from a relative newbie: What is this `3_u16` syntax? I have seen the `3u16` syntax but never with the underscore. Is there a difference between this syntax?

I would suggest that using `3u16` or even `let items: Vec<u16>= vec![3, 2, 1, 10];` to make it more clear for people that are not too experienced with Rust. But that is just my humble opinion!

Reply:
The `_` here is merely an effect of separators in numeric literals, e.g. `3_000_000_u32` or `3_000____u32`. The underscores are for readability and have no effect otherwise. `3u16` and `3_u16` are thus equivalent.

Reply:
Alright! Thanks for sharing! :)